### ISSUE TITLE
Cpp 17 issue with FairMQ 1.4.8  und Apple clang version 12.0.0 (clang-1200.0.26.2)

### ISSUE THREAD

**Issue by MohammadAlTurany** [FairRootGroup/FairMQ#298]
**Describe the bug**
FairMQ does not compile with Apple clang version 12.0.0 (clang-1200.0.26.2)
 
**To Reproduce**

MacOS 10.15.6
Apple clang version 12.0.0 (clang-1200.0.26.2)
FairMQ 1.4.8 or later 

**Output**
```

/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/memory:1720:13: error: static_assert failed due to requirement
      'is_cpp17_move_insertable<std::1::allocator<FairMQChannel>, false>::value' "The specified type does not meet the requirements of Cpp17MoveInsertable"
            static_assert(is_cpp17_move_insertable<allocator_type>::value,
            ^             ~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/vector:952:21: note: in instantiation of function template specialization
      'std::1::allocator_traits<std::__1::allocator<FairMQChannel> >::construct_backward_with_exception_guarantees<FairMQChannel *>' requested here
    alloc_traits::construct_backward_with_exception_guarantees(
                    ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/vector:1668:5: note: in instantiation of member function 'std::1::vector<FairMQChannel,
      std::__1::allocator<FairMQChannel> >::swap_out_circular_buffer' requested here
    swap_out_circular_buffer(v);
    ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/vector:1686:9: note: in instantiation of function template specialization
      'std::1::vector<FairMQChannel, std::__1::allocator<FairMQChannel> >::emplace_back_slow_path<const std::1::basic_string<char> &, int &,
      std::1::map<std::1::basic_string<char>, boost::any, std::1::less<std::1::basic_string<char> >, std::1::allocator<std::1::pair<const
      std::__1::basic_string<char>, boost::any> > > >' requested here
        emplace_back_slow_path(_VSTD::forward<_Args>(args)...);
        ^
/Users/turany/development/FairSoft/legacy/basics/FairMQ/fairmq/FairMQDevice.cxx:285:35: note: in instantiation of function template specialization
      'std::1::vector<FairMQChannel, std::1::allocator<FairMQChannel> >::emplace_back<const std::__1::basic_string<char> &, int &,
      std::1::map<std::1::basic_string<char>, boost::any, std::1::less<std::1::basic_string<char> >, std::1::allocator<std::1::pair<const
      std::__1::basic_string<char>, boost::any> > > >' requested here
            fChannels[info.first].emplace_back(info.first, i, fConfig->GetPropertiesStartingWith(tools::ToString("chans.", info.first, ".", i, ".")));
                                  ^
1 error generated.
make[2]: *** [fairmq/CMakeFiles/FairMQ.dir/FairMQDevice.cxx.o] Error 1
```

**Comment by aphecetche** [comment]
@MohammadAlTurany any news on this one ? I've made the mistake to let the upgrade to Xcode 12 happen on my Mac and now I'm hitting this one ;-)

**Comment by dennisklein** [comment]
When @MohammadAlTurany hit this, Xcode 12 was still in beta. We decided to revisit, once the final Xcode 12 is out. Is this now the case? Even if not, we might want to increase priority on this one. AFAIK there are no news.

**Comment by aphecetche** [comment]
@dennisklein yes Xcode 12 has been released last week (and installed on Macs by some people -like me- not paying too much attention to the major bump :-( )

**Comment by rbx** [comment]
I'll install the update and check.

**Comment by mattkretz** [comment]
For reference after @rbx and I chatted: [`emplace_back` requires *Cpp17MoveInsertable*](http://eel.is/c++draft/sequence.reqmts#tab:container.seq.opt).

**Comment by rbx** [comment]
The issue is now fixed in dev and master. New tag is available with the fix - v1.4.24.

**Comment by aphecetche** [comment]
Thanks for the quick fix !
